addons/payment/models/payment_transaction.py

```diff
@@ -631,73 +631,93 @@ def _process_notification_data(self, notification_data):
         """
         self.ensure_one()
 
-    def _set_pending(self, state_message=None):
+    def _set_pending(self, state_message=None, extra_allowed_states=None):
         """ Update the transactions' state to `pending`.
 
         :param str state_message: The reason for setting the transactions in the state `pending`.
+        :param list[str] extra_allowed_states: Additional allowed states specific to a provider on
+                                               some flows
         :return: The updated transactions.
         :rtype: recordset of `payment.transaction`
         """
         allowed_states = ('draft',)
         target_state = 'pending'
-        txs_to_process = self._update_state(allowed_states, target_state, state_message)
+        txs_to_process = self._update_state(
+            allowed_states, extra_allowed_states, target_state, state_message
+        )
```

Comment:
```suggestion
        txs_to_process = self._update_state(
            allowed_states + extra_allowed_states, target_state, state_message
        )
```
So you can handle only one parameter in `_update_state`.